src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js

```diff
@@ -1,360 +1,42 @@
-
-    if (this.options.visible) {
-      L.TileLayer.prototype.setOpacity.call(this, this.options.opacity);
-      this.fire('updated');
-    }
-  },
-
-  /**
-   * When Leaflet adds the layer... go!
-   * @params {map}
-   */
-  onAdd: function (map) {
-    var self = this;
-    this.options.map = map;
-  },
-
-  /**
-   * When removes the layer, destroy interactivity if exist
-   */
-  onRemove: function (map) {
-    if (this.options.added) {
-      this.options.added = false;
-      L.TileLayer.prototype.onRemove.call(this, map);
-    }
-  },
-
-  /**
-   * Update CartoDB layer
-   * generates a new url for tiles and refresh leaflet layer
-   * do not collide with leaflet _update
-   */
-  __update: function (done) {
-    var self = this;
-    this.fire('updated');
-    this.fire('loading');
-    var map = this.options.map;
-
-    var tilejson = self.model.get('urls');
-    if (tilejson) {
-      self.tilejson = tilejson;
-      self.setUrl(self.tilejson.tiles[0]);
-      // self.setProvider({
-      //   styles: [],
-      //   templateUrl: self.tilejson.tiles[0]
-      // });
-      // manage interaction
-      self._reloadInteraction();
-      self.ok && self.ok();
-      done && done();
-    } else {
-      self.error && self.error('URLs have not been fetched yet');
-      done && done();
-    }
-  },
-
-  /**
-   * Bind events for wax interaction
-   * @param {Object} Layer map object
-   * @param {Event} Wax event
-   */
-  _manageOnEvents: function (map, o) {
-    var layer_point = this._findPos(map, o);
-
-    if (!layer_point || isNaN(layer_point.x) || isNaN(layer_point.y)) {
-      // If layer_point doesn't contain x and y,
-      // we can't calculate event map position
-      return false;
-    }
-
-    var latlng = map.layerPointToLatLng(layer_point);
-    var event_type = o.e.type.toLowerCase();
-    var screenPos = map.layerPointToContainerPoint(layer_point);
-
-    switch (event_type) {
-      case 'mousemove':
-        if (this.options.featureOver) {
-          return this.options.featureOver(o.e, latlng, screenPos, o.data, o.layer);
-        }
-        break;
-
-      case 'click':
-      case 'touchend':
-      case 'touchmove': // for some reason android browser does not send touchend
-      case 'mspointerup':
-      case 'pointerup':
-      case 'pointermove':
-        if (this.options.featureClick) {
-          this.options.featureClick(o.e, latlng, screenPos, o.data, o.layer);
-        }
-        break;
-      default:
-        break;
-    }
+      var index = child.get('order') - 1;
+      this.setCartoCSS(index, style);
+    }, this);
   },
 
-  /**
-   * Bind off event for wax interaction
-   */
-  _manageOffEvents: function (map, o) {
-    if (this.options.featureOut) {
-      return this.options.featureOut && this.options.featureOut(o.e, o.layer);
-    }
-  },
+  _onTileJSONChanged: function () {
+    var tilejson = this.model.get('urls');
+    this.setUrl(tilejson.tiles[0]);
```

Comment:
@fdansv IMO the following chunk could/should be part of the implementation of `setURL`:

```

    this.options.styles = this.model.layers.pluck('cartocss');

    if (this.renderers.length === 0) {
      L.CartoDBd3Layer.prototype.onAdd.call(this, this.leafletMap);
    } else {
      this.setProvider({
        styles: this.model.layers.pluck('cartocss'),
        urlTemplate: tilejson.tiles[0]
      });
    }

    this.options.styles = this.model.layers.pluck('cartocss');

    if (this.renderers.length === 0) {
      L.CartoDBd3Layer.prototype.onAdd.call(this, this.leafletMap);
    } else {
      this.setProvider({
        styles: this.model.layers.pluck('cartocss'),
        urlTemplate: tilejson.tiles[0]
      });
    }
```

What do you think?